stock_warehouse_flow/models/stock_warehouse_flow.py

```diff
@@ -385,6 +363,68 @@ def _get_rule_from_delivery_route(self, html_exc=False):
             )
         return rule
 
+    def _is_valid_for_move(self, move):
+        self.ensure_one()
+        if not self.move_domain:
+            return move
+        domain = safe_eval(self.move_domain or "[]")
+        if not domain:
+            return move
+        return move.filtered_domain(domain)
+
+    def _prepare_move_split_vals(self, move, split_qty):
+        split_qty_uom = move.product_id.uom_id._compute_quantity(
+            split_qty, move.product_uom, rounding_method="HALF-UP"
+        )
+        return move._prepare_move_split_vals(split_qty_uom)
+
+    def _split_move(self, move, split_qty):
+        split_move = move.copy(self._prepare_move_split_vals(move, split_qty))
+        new_product_qty = move.product_id.uom_id._compute_quantity(
+            move.product_qty - split_qty, move.product_uom, round=False
+        )
+        move.product_uom_qty = new_product_qty
+        return split_move
+
+    def _compute_split_qty_by_packaging(self, move, packaging):
+        """Returns the qty to split for move and packaging"""
+        rounding = move.product_id.uom_id.rounding
+        if (
+            packaging.packaging_type_id not in self.packaging_type_ids
+            or float_compare(
+                packaging.qty, move.product_qty, precision_rounding=rounding
+            )
+            > 0
+        ):
+            return
+        pack_qty = int(move.product_qty / packaging.qty) * packaging.qty
+        split_qty = move.product_qty - pack_qty
+        # There is nothing to split if the split_qty is 0
+        # then the move qty is the same or a multiple of the packaging qty
+        if float_compare(split_qty, 0, precision_rounding=rounding) <= 0:
+            return
+        return split_qty
+
+    def _split_move_by_packaging(self, move, packaging):
+        self.ensure_one()
+        move.ensure_one()
+        packaging.ensure_one()
+        split_qty = self._compute_split_qty_by_packaging(move, packaging)
+        if split_qty:
+            return self._split_move(move, split_qty)
+
+    def split_move(self, move):
+        """Try to split the move by the first matching packaging"""
+        self.ensure_one()
+        if not self.split_by_packaging_qty or not self.packaging_type_ids:
+            return
+
+        # Move gets split by first matching packaging
+        for packaging in move.product_id.packaging_ids:
+            split_move = self._split_move_by_packaging(move, packaging)
+            if split_move:
+                return split_move
```

Comment:
Maybe the split move could be split again with another packaging. I would continue looping